compiler/src/dotty/tools/dotc/Compiler.scala

```diff
@@ -35,10 +35,10 @@ class Compiler {
   protected def frontendPhases: List[List[Phase]] =
     List(new Parser) ::             // Compiler frontend: scanner, parser
     List(new TyperPhase) ::         // Compiler frontend: namer, typer
-    List(new CheckUnused.PostTyper) ::  // Check for unused elements
     List(new YCheckPositions) ::    // YCheck positions
     List(new sbt.ExtractDependencies) :: // Sends information on classes' dependencies to sbt via callbacks
     List(new semanticdb.ExtractSemanticDB) :: // Extract info into .semanticdb files
+    List(new CheckUnused.PostTyper, new CheckShadowing) :: // Check for unused elements and shadowing elements
```

Comment:
For having both `CheckUnused` and `CheckShadowing` `MiniPhase`s in the same `MegaPhase` it's necessary to set their common `MegaPhase` after the `ExtractSemanticDB` one.

Otherwise it will triggers an error in `scala3-compiler-bootstrapped/testOnly dotty.tools.dotc.semanticdb.SemanticdbTests`
I didn't figured out exactly why, I remarked that the error comes from an higher level as when getting rid of all the traversing logic in `CheckUnused` and in `CheckShadowing` the error was still present.

Hence, just having two `MiniPhase`s aggregated seems to influence the `ExtractSemanticDB` phase.
For still having the "linting" phases just after the typer while avoiding that weird behavior, one could also have them in two separate `MegaPhase`s, it works well.

Reply:
Let's keep them as two MegaPhases in that case and leave CheckUnused untouched. We will look into that later.